### ISSUE TITLE
[HttpClient] Closing channel pool is not checking ChannelFuture status

### ISSUE THREAD

**Issue by pojaszek** [ratpack/ratpack#1348]
Using ratpack in 1.5.4 version

DefaultHttpClient is responsible for closing channel pool:
```
@Override
  public void close() {
    channelPoolMap.close();
  } 
```
for each key it invokes remove method in AbstractChannelPoolMap:
```
public final boolean remove(K key) {
        P pool =  map.remove(checkNotNull(key, "key"));
        if (pool != null) {
            pool.close();
            return true;
        }
        return false;
    }
```
And in SimpleChannelPool:
```
@Override
    public void close() {
        for (;;) {
            Channel channel = pollChannel();
            if (channel == null) {
                break;
            }
            channel.close();
        }
    }
```
channel.close() returns ChannelFuture object but no one is checking whether the operation completes.

This causes a problem that connection pool is not closed before the ratpack is shutdown. We are receiving a lot of errors after redeploying ratpack service::
```
java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:844)
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:328)
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:321)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:769)
	at io.netty.util.concurrent.DefaultPromise.safeExecute(DefaultPromise.java:764)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:432)
	at io.netty.util.concurrent.DefaultPromise.setFailure(DefaultPromise.java:112)
	at io.netty.channel.DefaultChannelPromise.setFailure(DefaultChannelPromise.java:89)
	at io.netty.channel.AbstractChannelHandlerContext.safeExecute(AbstractChannelHandlerContext.java:1010)
	at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:610)
	at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:465)
	at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:983)
	at io.netty.channel.AbstractChannel.close(AbstractChannel.java:238)
	at io.netty.channel.pool.SimpleChannelPool.close(SimpleChannelPool.java:397)
	at io.netty.channel.pool.FixedChannelPool.access$1201(FixedChannelPool.java:38)
	at io.netty.channel.pool.FixedChannelPool$5.run(FixedChannelPool.java:458)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasksFrom(SingleThreadEventExecutor.java:380)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:355)
	at io.netty.util.concurrent.SingleThreadEventExecutor.confirmShutdown(SingleThreadEventExecutor.java:701)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:323)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at ratpack.exec.internal.DefaultExecController$ExecControllerBindingThreadFactory.lambda$newThread$0(DefaultExecController.java:136)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

```

Worth to say that the above error is not happening after adding a Thread sleep for a few seconds before the ratpack is shutting down.

**Comment by jdbevan** [comment]
Is this fixed by https://github.com/netty/netty/pull/7927 ?

**Comment by beckje01** [comment]
When 4.1.26 is out we will update

**Comment by jdbevan** [comment]
Great, thanks

**Comment by johnrengelman** [comment]
This should be fixed with the newer netty version in 1.6.0-rc-1
